minidump/src/minidump.rs

```diff
@@ -1648,6 +1715,68 @@ impl<'a> MinidumpStream<'a> for MinidumpMemoryList<'a> {
     }
 }
 
+impl<'a> MinidumpStream<'a> for MinidumpMemory64List<'a> {
+    const STREAM_TYPE: MINIDUMP_STREAM_TYPE = MINIDUMP_STREAM_TYPE::Memory64ListStream;
+
+    fn read(
+        bytes: &'a [u8],
+        all: &'a [u8],
+        endian: scroll::Endian,
+    ) -> Result<MinidumpMemory64List<'a>, Error> {
+        let mut offset = 0;
+        let u: u64 = bytes
+            .gread_with(&mut offset, endian)
+            .or(Err(Error::StreamReadFailure))?;
+
+        let mut rva: u64 = bytes
+            .gread_with(&mut offset, endian)
+            .or(Err(Error::StreamReadFailure))?;
+
+        let (count, counted_size) = ensure_count_in_bound(
+            bytes,
+            u as usize,
```

Comment:
This cast is the only suspicious thing I found here.

Because if we're on a 32 bit platform (which i'm going to assume is reasonable, but a 16 bit platform is not, people are probably not parsing minidumps on their SNES and if they are I would like to meet them), then a valid dump could actually lose information here, and we'll think count is shorter than it actually is.

But that's fine, actually, because on any platform, if the file has more than *our* usize's count of elements, we couldn't have even got here, because we can fit the full dump in memory.

So it's Fine Actually but it's subtle as to *how* it's fine.

That being said, if we see a dump that's got an invalid count of elements, it will selectively pass/fail based on which platform we're on (I can craft a dump that only fails on 64 bit platforms), which seems non-ideal.

TLDR: I'd make this a checked try_into, returning an error if it's out of bounds. That won't introduce any failures on valid files, but would make invalid files fail regardless of the platform's pointer size.

Maybe this method should be changed to internally do the conversion, and take `TryInto<usize>` so you don't need to do it yourself.

Up to @luser though.